source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/locale-select/index.tsx

```diff
@@ -22,39 +22,54 @@ export type IProps = IOwnProps & WithDataProps & i18nProps;
 const translationMap = { 'en-US': 'en', 'es-419': 'es', 'fr-FR': 'fr' };
 
 class LocaleSelect extends React.Component<IProps> {
+  defaultValue = '';
+
+  componentDidMount() {
+    this.defaultToFirst();
+  }
+
+  componentDidUpdate() {
+    this.defaultToFirst();
+  }
+
+  defaultToFirst = () => {
+    const { onChange, selected } = this.props;
+    if (!selected) {
+      onChange(this.defaultValue);
+    }
+  };
+
```

Comment:
as a function-component, this could be converted like:
```ts
export default function LocaleSelect({ selected, onChange}: IProps) {
  const onSelect = async (e, { value }) => {
    e.preventDefault();

     if (value === selected) {
       return;
     }

     onChange(value);
  }

  // ...

  return (
    <Dropdown defaultValue={selected} ...otherstuff />
   );
}
```

Though, looking at those closer, I must ask: what is the point of the default value property?

Reply:
I found that I could significantly simplify this class.  It now has a first time variable so that it can set the value of the variable in the caller for the case where no selection is made initially, which is kind of what it was trying to do before with the default value.